File: php/src/Amen/Resources/Withdrawals.php
<?php
declare(strict_types=1);
namespace Amen\Resources;
use Amen\AmenClient;
final class Withdrawals
{
    public function __construct(private AmenClient $c) {}
    public function create(string $bankAccountId, string $amount): array { return $this->c->request('POST', '/withdrawals/', ['json' => ['bank_account_id' => $bankAccountId, 'amount' => $amount]]); }
    public function get(string $n): array { return $this->c->request('GET', "/withdrawals/$n"); }
    public function list(array $params = []): array { return Page::from($this->c->request('GET', '/withdrawals/', ['params' => $params]), 'withdrawals'); }
}

File: kotlin/amen-client/src/main/kotlin/sa/amnn/kit/resources/Withdrawals.kt
package sa.amnn.kit.resources

import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.put
import sa.amnn.kit.AmenClient
import sa.amnn.kit.Withdrawal

class Withdrawals(private val c: AmenClient) {
    suspend fun create(bankAccountId: String, amount: String) = c.request("POST", "/withdrawals/", Withdrawal.serializer(), body = buildJsonObject { put("bank_account_id", bankAccountId); put("amount", amount) })!!
    suspend fun get(n: String) = c.request("GET", "/withdrawals/$n", Withdrawal.serializer())!!
    suspend fun list(params: Map<String, String?> = emptyMap()) = c.page("/withdrawals/", params, "withdrawals", Withdrawal.serializer())
}
